go/vt/mysqlctl/schema.go

```diff
@@ -244,9 +244,11 @@ func (mysqld *Mysqld) normalizedSchema(ctx context.Context, dbName, tableName, t
 	norm := qr.Rows[0][1].ToString()
 	norm = autoIncr.ReplaceAllLiteralString(norm, "")
 	if tableType == tmutils.TableView {
-		// Views will have the dbname in there, replace it
-		// with {{.DatabaseName}}
-		norm = strings.Replace(norm, backtickDBName, "{{.DatabaseName}}", -1)
+		// Views will have the dbname qualifier in there, we remove it
+		// so that the results for Views match:
+		//   1. The SHOW CREATE VIEW output from MySQL
+		//   2. The CREATE TABLE output in these results, which does not have the DB name qualifier
+		norm = strings.Replace(norm, backtickDBName+".", "", -1)
```

Comment:
I'm wary of `strings.Replace()` as means to manipulate queries. What happens if there's a column that has same name as the database? At the very least I'd say lets replace `-1` with `1`:

```suggestion
		norm = strings.Replace(norm, backtickDBName+".", "", 1)
```

but then, when I experiment on my local machine:
```sql
master [localhost:21324] {msandbox} (test) > show create table `test`.`v` \G
*************************** 1. row ***************************
                View: v
         Create View: CREATE ALGORITHM=UNDEFINED DEFINER=`msandbox`@`localhost` SQL SECURITY DEFINER VIEW `v` AS select `t`.`id` AS `id`,`t`.`i` AS `i`,`t`.`tx` AS `tx` from `t`
character_set_client: utf8mb4
collation_connection: utf8mb4_0900_ai_ci
```
I don't see a database qualifier in the resulting `CREATE VIEW` statement. therefore, should there be a column named `test`, it will be wrongly replaced.

I realize the bug predates this PR. But then, I also see no testing to this logic. Should we seize the opportunity to fix this? We can e.g. `sqlparser.ParseStatement` into a `CreateView`,  then remove the schema qualifier, then export back via `sqlparser.String()`.

WDYT?

Reply:
This is about the `vtctl GetSchema` results and the RPC it uses. You can see an example if you exapand `Here's a demonstration:` in the description. 

I had similar concerns about the potential for problems, but since we already had that issue in this code I set it aside. I can explore alternatives though, I'm going to move this back to draft anyway as I'm adding a new e2e test for `GetSchema` in `vitess.io/vitess/go/test/endtoend/tabletmanager`.

Reply:
got it. I actually missed the `Here's a demonstration:` section.

I agree that the solution in this PR is better than already existing code.